### DISCUSSION TITLE
How can I use clang++ with cc_library on Windows?

### DISCUSSION THREAD

**Discussion by theoparis** [bazelbuild/bazel#24057] [Q&A]
I tried various things - setting CXX/CC, adding --repo_env=CC=clang to .bazelrc, etc. and `bazel build //...` in a minimal project on Windows keeps trying to use msvc's cl.exe 😕

**Comment by fmeum** [comment]
Have you tried to follow https://bazel.build/configure/windows#clang? For current Bazel versions, you need to follow the instructions for `--incompatible_enable_cc_toolchain_resolution`.

**Comment by malkia** [comment]
You can find several projects using "clang-cl" for example - https://github.com/search?q=%22clang-cl%22+path%3A.bazelrc&ref=opensearch&type=code - not sure if this is exactly what you want - e.g. "clang-cl", which tries to resemble most of what MSVC would do - https://clang.llvm.org/docs/MSVCCompatibility.html - I've been using it for my projects, because MSVC (heh), can't handle 260+ char paths, and as it happens with bazel some paths are really long... Though `clang-cl` can handle it!
